src/classes/UnitOfWork.cls

```diff
@@ -0,0 +1,147 @@
+public class UnitOfWork extends NebulaCore {
```

Comment:
I like the looks of this!

Reply:
Awww, yiss! I started this months ago, but haven't had a chance to wrap it up yet. Hopefully I'll get back to it in the next few weeks